src/db.c

```diff
@@ -1661,6 +1669,59 @@ db_purge_cruft(time_t ref)
 #undef Q_TMPL
 }
 
+void
+db_purge_source_cruft(time_t ref, const char *source)
+{
+#define Q_TMPL "DELETE FROM directories WHERE id >= %d AND db_timestamp < %" PRIi64 " AND source = %Q;"
```

Comment:
I guess you already considered it, but are you sure it is a good idea to have the source as a string and not a number/an id? To me it looks a bit off. There would be a small performance gain from using an id instead, but I am mostly thinking that it might be easier going ahead with just an id. Not that I can think of any concrete issues that the string gives. And I see below that you added a "description", which seems to serve the purpose of mapping to something more human readable.

Reply:
I have to admit, I did not give it much thought. I started with just moving the existing "name" into the DB and I kind of like to see directly in the DB what a value means (enums make it a bit harder to understand what meaning the values have). But I agree, that an int/enum makes more sense (and better fits to the existing data_kind / media_kind columns).

With an int there is now no NULL value. I use `0` for that case. I am not sure, if this is the best way to do it ...

Reply:
To me 0 sounds fine.